sdk/core/System.ClientModel/src/Convenience/ReturnWhen.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+
+namespace System.ClientModel;
+
+/// <summary>
+/// Indicates whether a client method that starts a long-running operation
+/// should return after the operation has started or after the service has
+/// completed processing of the operation.
+/// </summary>
+public enum ReturnWhen
+{
+    /// <summary>
+    /// Indicates the client method should wait to return until the service has
+    /// completed processing of the operation.
+    /// </summary>
+    Completed,
+
+    /// <summary>
+    /// Indicates the client method should return after the service has responded
+    /// to the request to start the operation.
+    /// </summary>
+    /// <remarks>When <see cref="Started"/> is passed to a client method that
+    /// creates an <see cref="OperationResult"/>, the caller must use
+    /// <see cref="OperationResult.WaitForCompletion"/> or other method to wait
+    /// for the operation to complete before being able to obtain the result of
+    /// the service operation.</remarks>
```

Comment:
This is helpful context - would it be useful to provide similar guidance for `Completed` e.g. the expected state of the `OperationResult.IsCompleted` and the availability of `Value`?

Reply:
Happy to add something similar here.